dot/rpc/modules/author_integration_test.go

```diff
@@ -395,6 +385,108 @@ func TestAuthorModule_HasKey_Integration(t *testing.T) {
+
+	const sr25519Seed = "0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a"
+	const sr25519Pubk = "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d"
+
+	insertSessionKeys := []struct {
+		ktype      []string
+		seed, pubk string
+	}{
+		{
+			ktype: []string{"gran"},
+			seed:  granSeed,
+			pubk:  granPubK,
+		},
+		{
+			ktype: []string{"babe", "imon", "audi"},
+			seed:  sr25519Seed,
+			pubk:  sr25519Pubk,
+		},
+	}
+
+	for _, toInsert := range insertSessionKeys {
+		for _, keytype := range toInsert.ktype {
+			err := auth.InsertKey(nil, &KeyInsertRequest{
+				Type:      keytype,
+				Seed:      toInsert.seed,
+				PublicKey: toInsert.pubk,
+			}, nil)
+			require.NoError(t, err)
+		}
+	}
+
+	testcases := map[string]struct {
+		pubSessionKeys string
+		expect         bool
+		waitErr        error
+	}{
+		"public keys are in the right order, should return true": {
+			pubSessionKeys: "0x6b802349d948444d41397da09ec597fbd8ae8fdd3dfa153b2bb2bddcf020457c" + // gran
+				"d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d" + // babe
+				"d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d" + // imon
+				"d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d", // audi
+			expect: true,
+		},
+		"uknown public keys in the right order, should return false": {
+			pubSessionKeys: "0x740550da19ef14023ea3e903545a6700160a55be2e4b733b577c91b053e38b8d" + // gran
+				"de6fa0da51c52cc117d77aeb329595b15070db444e7ed4c4adec714b291c1845" + // babe
+				"de6fa0da51c52cc117d77aeb329595b15070db444e7ed4c4adec714b291c1845" + // imon
+				"de6fa0da51c52cc117d77aeb329595b15070db444e7ed4c4adec714b291c1845", // audi
+			expect: false,
+		},
+		"public keys are not in the right order, should return false": {
+			pubSessionKeys: "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d" + // gran
+				"6b802349d948444d41397da09ec597fbd8ae8fdd3dfa153b2bb2bddcf020457c" + // babe
+				"d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d" + // imon
+				"d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d", // audi
+			expect: false,
+		},
+		"incomplete keys": {
+			pubSessionKeys: "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d" + // gran
+				"6b802349d948444d41397da09ec597fbd8ae8fdd3dfa153b2bb2bddcf020457c", // babe
+			expect: false,
+		},
+		"empty public keys": {
+			pubSessionKeys: "", // babe
+			expect:         false,
+			waitErr:        errors.New("invalid string"),
+		},
+	}
+
+	for tname, tt := range testcases {
+		tt := tt
+		t.Run(tname, func(t *testing.T) {
+			t.Parallel()
+			var req HasSessionKeyRequest
+			req.PublicKeys = tt.pubSessionKeys
+
+			var res HasSessionKeyResponse
+
+			err := auth.HasSessionKeys(nil, &req, &res)
+
+			if tt.waitErr != nil {
+				require.EqualError(t, tt.waitErr, err.Error())
+				require.False(t, bool(res))
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tt.expect, bool(res))
```

Comment:
nit so we have only one assert for res

```suggestion
			} else {
				require.NoError(t, err)
			}
			require.Equal(t, tt.expect, bool(res))
```

Reply:
Changed!